google-oauth-client/src/main/java/com/google/api/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -206,9 +223,20 @@ static TokenResponse requestAccessToken(AuthorizationCodeFlow flow, String code)
    * @param authorizationCode authorization code.
    */
   public AuthorizationCodeTokenRequest newTokenRequest(String authorizationCode) {
+    HttpExecuteInterceptor pkceClientAuthenticationWrapper = new HttpExecuteInterceptor() {
+      @Override
+      public void intercept(HttpRequest request) throws IOException {
+        clientAuthentication.intercept(request);
+        if (pkce != null) {
+          Map<String, Object> data = Data.mapOf(UrlEncodedContent.getContent(request).getData());
+          data.put("code_verifier", pkce.getVerifier());
```

Comment:
Do we need to re-set the data back on the request? Or is this somehow mutating the data in place?

Reply:
My understanding is that the request is used only once for the request to obtain the authorization code token.

So I "borrowed" the implementation that is used by the [ClientsParameterAuthentication](https://github.com/googleapis/google-oauth-java-client/blob/6447917c657a5ae4267afbab74dfdb890bbfbf28/google-oauth-client/src/main/java/com/google/api/client/auth/oauth2/ClientParametersAuthentication.java#L98) class.

As I said, I believe this object is only used for the one request to get the authorization code and therefore I believe we don't need to reset anything. I'll see if I can perform some experiments to make sure that subsequent requests don't leak this information.

Reply:
I experimented a little bit with the `PKCESample` class and after requesting authorization and then creating a request via a transport obtained from the credential object received from the authorization code flow, the code_verifier header is no longer in any requests that are sent out.

So unless I'm misunderstanding your comment, I think we're good.